joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/parser/AntlrParser.scala

```diff
@@ -188,3 +183,124 @@ class ResourceManagedParser(clearLimit: Double, debug: Boolean = false, profilin
 
   override def close(): Unit = clearDFA()
 }
+
+class ParserProfiler {
+
+  private val logger                    = LoggerFactory.getLogger(getClass)
+  private val ruleCost                  = TrieMap.empty[String, RuleTimeCost]
+  private val fileCost                  = TrieMap.empty[String, Long]
+  private var projectRoot: Option[Path] = None
+
+  sys.addShutdownHook {
```

Comment:
Hmm, I wonder if a shutdown hook is really the best option here - feels a bit messy, since we're mixing concerns: the lifetime of the jvm vs. the completion of a scan. 
Sure, that's identical in all current use cases, but are we certain that's going to stay like that forever? 
I'd feel better if we collect the results somewhere, maybe even pass them properly, and explicitly print them at the end of the analysis.

Reply:
Not sure I understand. If a scan doesn't finish within a couple of minutes, how does a shutdown hook help with that?

Reply:
If we, for example, want to find out a summary of the most expensive rules so far (and not wait for hours), and then we use `Ctrl+C` to prematurely stop `joern-parse`, we can still get the results.

Reply:
@mpollmeier Just following up on this. If I remove the hook, it's not a big issue, as the conclusion of which rules are expensive are pretty universal across repos. Would that be preferrable?

Reply:
Ah ok, understood. Given that for current use cases the hook is not a problem, we can just leave it there and add a comment saying it's optional and just for debug output and can be removed if need be. Or something like that...